instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/handler.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsSdk
+      # Generates Spans for all interactions with AwsSdk
+      class Handler < Seahorse::Client::Handler
+        def call(context)
+          span_name, attributes = calculate_span(context)
+
+          tracer.in_span(span_name, kind: OpenTelemetry::Trace::SpanKind::CLIENT, attributes: attributes) do |span|
+            execute = proc {
+              super(context).tap do |response|
+                if (err = response.error)
+                  span.record_exception(err)
+                  span.status = Trace::Status.error(err)
+                end
+              end
+            }
+
+            if instrumentation_config[:suppress_internal_instrumentation]
+              OpenTelemetry::Common::Utilities.untraced(&execute)
+            else
+              execute.call
+            end
+          end
+        end
+
+        def calculate_span(context)
+          service_name = context.client.class.api.metadata['serviceId'] || context.client.class.to_s.split('::')[1]
+          span_name = "#{service_name}.#{context.operation.name}"
+          attributes = {
+            'aws.region' => context.config.region,
```

Comment:
The RPC semantic conventions comply with the specification: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md, however, the `aws.region` semantic convention isn't formally specified. OTel JS [uses it](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-aws-sdk#span-attributes) though.

I am wondering if the Resource semantic convention of setting `cloud.provider` and `cloud.region` would be a better option (see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/cloud.md#cloud). Happy to have others chime in on the best approach here.

Reply:
You're raising a good question. 

I'm not sure resource attributes are a good fit for this attribute. 
As I understand, resource attributes are immutable representation of the running service.
For example, if I'm running a service on EC2, `cloud.provider` and `cloud.region` resource attributes should keep the value of the running environment. If the running service is producing `sqs.send_message` operation to a queue in a different region, `cloud.region` should have the EC2's region and not the SQS destination region.

Maybe we should use `cloud.destination.region` instead of `aws.region`.

The cloud semantic conventions are still in Experimental status. This topic is not covered yet. 
IMO we can keep the behavior as implemented in OTEL JS, and update it in the future when it will have a clear definition.

WDYT?

Reply:
my 2 cents is, unless we want to put in the effort to make an otep or spec change upstream , stick with what js does. i think `cloud.destination.region` is a completely reasonable choice fwiw, but I just think practically speaking lets lean on the experimental status here and let ppl speak up if they hate the attr name.

Reply:
i'm also fine with sticking `aws.region` as is for now.